packages/apm/src/integrations/tracing.ts

```diff
@@ -852,17 +852,18 @@ export class Tracing implements Integration {
     }
 
     const count = Object.keys(Tracing._activities).length;
-
-    Tracing._log('[Tracing] activies count', count);
+    Tracing._log('[Tracing] activities count', count);
 
     if (count === 0 && Tracing._activeTransaction) {
       const timeout = Tracing.options && Tracing.options.idleTimeout;
       Tracing._log(`[Tracing] Flushing Transaction in ${timeout}ms`);
       // We need to add the timeout here to have the real endtimestamp of the transaction
-      // Remeber timestampWithMs is in seconds, timeout is in ms
+      // Remember timestampWithMs is in seconds, timeout is in ms
       const end = timestampWithMs() + timeout / 1000;
       setTimeout(() => {
-        Tracing.finishIdleTransaction(end);
+        if (Object.keys(Tracing._activities).length === 0 && Tracing._activeTransaction) {
+          Tracing.finishIdleTransaction(end);
+        }
```

Comment:
We have to be careful here. I tinkered around with this for a long time.
While in your specific example it might improve the transaction, in other cases this has the potential of not sending the transaction at all.
Also, this change was made on purpose, before (see git history) we have a debouce timer that kept the Transaction alive which resulted in may weird very long Transaction. This change was made so after all the initial requests are done and the first "idle" occurs, we finish the Transaction no matter what (to prevent long transactions).
We finish Span that would have been started and mark them as `cancelled`.

Reply:
I see. The issue here is that right now the moment the activities goes to 0, the idle transaction ends.

I see what you mean by long running transaction (like if a page keeps creating single activities), but a user can easily just adjust the timeout option to instantly call finish idle transaction.

We could also apply a heuristic here, and say something like if the activities went from 0 -> 1 -> 0 -> 1 a certain amount of times, we should just end the idle transaction. 

This change should still lead to transactions finishing through the heartbeat though right?

Right now testing on multiple Sentry pages, a single page load fails to account for a lot of our requests, and I’d rather optimize for that use case then 0 -> 1 -> 0 ->1 causing long running transactions.